cmake/external/icu.cmake

```diff
@@ -0,0 +1,164 @@
+    set(ICU_STATIC_SUFFIX ".lib")
+    set(ICU_INSTALL_LIB_SUBDIR "lib64")
+    set(ICU_INSTALL_BIN_SUBDIR "bin64")
+    set(ICU_UC_LIB_NAME "icuuc")
+    set(ICU_I18N_LIB_NAME "icuin")
+    set(ICU_DATA_LIB_NAME "icudt")
+    set(ICU_UC_SHARED_LIB_NAME "${ICU_UC_LIB_NAME}${ICU_VERSION}")
+    set(ICU_I18N_SHARED_LIB_NAME "${ICU_I18N_LIB_NAME}${ICU_VERSION}")
+    set(ICU_DATA_SHARED_LIB_NAME "${ICU_DATA_LIB_NAME}${ICU_VERSION}")
+else()
+    set(ICU_SHARED_PREFIX ${CMAKE_SHARED_LIBRARY_PREFIX})
+    set(ICU_STATIC_PREFIX ${CMAKE_STATIC_LIBRARY_PREFIX})
+    set(ICU_SHARED_SUFFIX ${CMAKE_SHARED_LIBRARY_SUFFIX})
+    set(ICU_STATIC_SUFFIX ${CMAKE_STATIC_LIBRARY_SUFFIX})
+    set(ICU_INSTALL_LIB_SUBDIR "lib")
+    set(ICU_INSTALL_BIN_SUBDIR "lib")
+    set(ICU_UC_LIB_NAME "icuuc")
+    set(ICU_I18N_LIB_NAME "icui18n")
+    set(ICU_DATA_LIB_NAME "icudata")
+    set(ICU_UC_SHARED_LIB_NAME ${ICU_UC_LIB_NAME})
+    set(ICU_I18N_SHARED_LIB_NAME ${ICU_I18N_LIB_NAME})
+    set(ICU_DATA_SHARED_LIB_NAME ${ICU_DATA_LIB_NAME})
+    
+    # Calculate the number of cores using CMake
+    execute_process(COMMAND nproc OUTPUT_VARIABLE CMAKE_JOB_POOL_SIZE)
+    string(STRIP ${CMAKE_JOB_POOL_SIZE} CMAKE_JOB_POOL_SIZE)
+endif()
+
+set(ICU_INCLUDE_DIRS "${ICU_INSTALL_DIR}/include")
+
+set(ICU_STATIC_LIB_DIR "${ICU_INSTALL_DIR}/${ICU_INSTALL_LIB_SUBDIR}")
+set(ICU_SHARED_LIB_DIR "${ICU_INSTALL_DIR}/${ICU_INSTALL_BIN_SUBDIR}")
+
+foreach(build_type RELEASE DEBUG)
+  foreach(icu_target UC I18N DATA)
+    if(icu_target STREQUAL DATA)
+      set(lib_postfix ${CMAKE_RELEASE_POSTFIX})
+    else()
+      set(lib_postfix ${CMAKE_${build_type}_POSTFIX})
+    endif()
+    
+    set(ICU_${icu_target}_LIB_${build_type} "${ICU_STATIC_LIB_DIR}/${ICU_STATIC_PREFIX}${ICU_${icu_target}_LIB_NAME}${lib_postfix}${ICU_STATIC_SUFFIX}")
+    set(ICU_${icu_target}_SHARED_LIB_${build_type} "${ICU_SHARED_LIB_DIR}/${ICU_SHARED_PREFIX}${ICU_${icu_target}_SHARED_LIB_NAME}${lib_postfix}${ICU_SHARED_SUFFIX}")
+    list(APPEND ICU_LIBRARIES_${build_type} ${ICU_${icu_target}_LIB_${build_type}})
+    list(APPEND ICU_SHARED_LIBRARIES_${build_type} ${ICU_${icu_target}_LIB_${build_type}})
+  endforeach()
+endforeach()
+
+include(ExternalProject)
+
+if(WIN32)
+  ExternalProject_Add(
+    ${ICU_TARGET_NAME}
+    URL ${ICU_URL}
+    URL_HASH ${ICU_URL_HASH}
+    PREFIX ${THIRD_PARTY_PATH}
+    SOURCE_DIR ${ICU_SOURCE_DIR}
+    INSTALL_DIR ${ICU_INSTALL_DIR}
+    CONFIGURE_COMMAND msbuild ${ICU_SOURCE_DIR}\\source\\allinone\\allinone.sln /p:Configuration=${ICU_BUILD_TYPE} /p:Platform=x64 /t:i18n /t:uconv /t:makedata 
+    BUILD_COMMAND ""
+    INSTALL_COMMAND ${CMAKE_COMMAND} -E copy_directory ${ICU_SOURCE_DIR}/include ${ICU_INSTALL_DIR}/include && 
+                    ${CMAKE_COMMAND} -E copy_directory ${ICU_SOURCE_DIR}/lib64 ${ICU_INSTALL_DIR}/lib64 &&
+                    ${CMAKE_COMMAND} -E copy_directory ${ICU_SOURCE_DIR}/bin64 ${ICU_INSTALL_DIR}/bin64
+    BUILD_BYPRODUCTS ${ICU_LIBRARIES_RELEASE}
```

Comment:
why does not it depend on build type and `RELEASE` is hardcoded here?

Reply:
fixed